src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1400,16 +1353,25 @@ void Hole::onChanged(const App::Property* prop)
         if (ThreadType.isValid()) {
             type = ThreadType.getValueAsString();
             ThreadSize.setEnums(getThreadDesignations(ThreadType.getValue()));
-            if (type != "None") {
+            if (type != "Custom") {
                 findClosestDesignation();
             }
         }
 
-        if (type == "None") {
+        if (type == "Custom") {
+            ThreadPitch.setReadOnly(false);
+            ThreadDiameter.setReadOnly(false);

```

Comment:
readOnly attributes are set below, try to keep it in one place please

Reply:
This was on purpose. The alternative was adding `setReadOnly(true)` in all the `else if` clauses

Reply:
move it below `bool isCustom = type == "Custom";`
as
```
ThreadPitch.setReadOnly(!(isCustom && isThreaded));
ThreadDiameter.setReadOnly(!(isCustom && isThreaded));
```